repo_fetcher.py

```
@@ -8,78 +8,127 @@
+                if not self._gitlab_client:
+                    self._gitlab_client = gitlab.Gitlab('https://gitlab.com', private_token=self.gitlab_token)
+                project = self._gitlab_client.projects.get(f"{owner}/{repo_name}")
+                return {
+                    'platform': 'gitlab',
+                    'project': project,
+                    'owner': owner,
+                    'repo_name': repo_name
+                }
+            elif platform == 'bitbucket':
+                if not self.bitbucket_token:
+                    raise ValueError("Bitbucket token not provided")
```

Ruff (0.14.13)

Abstract `raise` to an inner function

(TRY301)